src/Compilers/CSharp/Test/Symbol/Symbols/Source/NullablePublicAPITests.cs

```diff
@@ -4200,5 +4200,91 @@ void M(out C c2)
+            comp.VerifyDiagnostics();
+            var tree = comp.SyntaxTrees[0];
+            var model = comp.GetSemanticModel(tree);
+            var assignmentsLefts = tree.GetRoot().DescendantNodes().OfType<AssignmentExpressionSyntax>().Select(a => a.Left).ToList();
+
+            Assert.Equal(2, assignmentsLefts.Count);
+            Assert.Equal(PublicNullableAnnotation.Annotated, model.GetTypeInfo(assignmentsLefts[0]).Type.NullableAnnotation);
+            Assert.Equal(PublicNullableAnnotation.NotAnnotated, model.GetTypeInfo(assignmentsLefts[1]).Type.NullableAnnotation);
+            Assert.Equal(PublicNullableAnnotation.Annotated, ((ILocalSymbol)model.GetSymbolInfo(assignmentsLefts[0]).Symbol).Type.NullableAnnotation);
+            Assert.Equal(PublicNullableAnnotation.Annotated, ((ILocalSymbol)model.GetSymbolInfo(assignmentsLefts[1]).Symbol).Type.NullableAnnotation);
+        }
+
+        [Fact]
+        [WorkItem(41488, "https://github.com/dotnet/roslyn/issues/41488")]
+        public void CompoundAssignmentLeftStateIsAfterAssignment()
+        {
+            var source = @"
+class C
+{
+    void M()
+    {
+        string? local = null;
+        local += string.Empty;
+    }
+}";
+
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics();
+            var tree = comp.SyntaxTrees[0];
+            var model = comp.GetSemanticModel(tree);
+            var localReference = tree.GetRoot().DescendantNodes().OfType<AssignmentExpressionSyntax>().Select(a => a.Left).Single();
+
+            Assert.Equal(PublicNullableAnnotation.NotAnnotated, model.GetTypeInfo(localReference).Type.NullableAnnotation);
+            Assert.Equal(PublicNullableAnnotation.Annotated, ((ILocalSymbol)model.GetSymbolInfo(localReference).Symbol).Type.NullableAnnotation);
+        }
+
+        [Fact]
+        [WorkItem(41488, "https://github.com/dotnet/roslyn/issues/41488")]
+        public void NullCoalescingAssignmentLeftStateIsAfterAssignment()
+        {
+            var source = @"
+class C
+{
+    void M()
+    {
+        string? local = null;
+        if (local != null)
+        {
+            local ??= null;
+            local ??= string.Empty;
+        }
+    }
+}";
+
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics();
+            var tree = comp.SyntaxTrees[0];
+            var model = comp.GetSemanticModel(tree);
+            var assignmentsLefts = tree.GetRoot().DescendantNodes().OfType<AssignmentExpressionSyntax>().Select(a => a.Left).ToList();
+
+            Assert.Equal(2, assignmentsLefts.Count);
+            Assert.Equal(PublicNullableAnnotation.Annotated, model.GetTypeInfo(assignmentsLefts[0]).Type.NullableAnnotation);
+            Assert.Equal(PublicNullableAnnotation.NotAnnotated, model.GetTypeInfo(assignmentsLefts[1]).Type.NullableAnnotation);
```

Comment:
Why is this `NotAnnotated`? The type of the expression is `string.Empty` which shold be annotated.

Reply:
Why would `string.Empty` be annotated?

Reply:
Okay if you're betting on that then don't use `string.Empty` because you're also using `CreateCompilation`. The contract for `CreateCompilation` is that we can, and will, silently move the underlying references forward as needed. Today they are roughly `netcoreapp2.1` which has no annotations. But in six months when I move them to `netcoreapp3.1` they will be annotated and this test will suddenly break. 

If you're looking for an API here which is unannotated to excercise this code path then I'd use `"#nullable disable` blocks to create. Otherwise the test will be fragile to future updates.

Reply:
@jaredpar I'm still confused. If `string.Empty` is from a disabled context, then we collapse that to `NotNull`. There's no fragility here.

Reply:
Ah yes because it's the value in context, not the annotation context of the member.